### PR TITLE
delete fclone() in RulerDAO

### PR DESCRIPTION
**Pull request by hchoura**

### CODE REVIEW

File: src/foam/nanos/ruler/RulerDAO.js [hunk 1]
```diff
@@ -140,7 +140,6 @@ if ( oldObj == null ) {
 // If the object was modified in an 'after' rule, then it is 'put'.
 FObject ret =  getDelegate().put_(x, obj);
 if ( ret != null ) {
-  ret = ret.fclone();
```
Comment (kgrgreer) [inline src/foam/nanos/ruler/RulerDAO.js:143]: Why was this here? Is it safe to remove? The applyRules can try to modify the object, but before the fclone() ret is frozen and can't be modified.

Comment (jlhughes) [reply src/foam/nanos/ruler/RulerDAO.js:143]: Rules are not responsible for 'putting' any changes. Also, if a rule does 'put', then any subsequent rules will receive the pre-put object.  
So rules should receive an un-frozen object which they can modify.
